### PR TITLE
Add support for Subaru Forester 2022

### PR DESCRIPTION
**Pull request by martinl**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: 7fd1e4f3a33c1673|2022-12-04--15-09-53
- [x] route with stock system: 7fd1e4f3a33c1673|2022-12-07--20-16-23

Prerequisites:
- https://github.com/commaai/opendbc/pull/758
- https://github.com/commaai/panda/pull/1187

### CODE REVIEW

File: selfdrive/car/subaru/carstate.py [hunk 1]
```diff
@@ -53,9 +53,13 @@ def update(self, cp, cp_cam, cp_body):
     steer_threshold = 75 if self.CP.carFingerprint in PREGLOBAL_CARS else 80
     ret.steeringPressed = abs(ret.steeringTorque) > steer_threshold
 
-    cp_cruise = cp_body if self.car_fingerprint in GLOBAL_GEN2 else cp
-    ret.cruiseState.enabled = cp_cruise.vl["CruiseControl"]["Cruise_Activated"] != 0
-    ret.cruiseState.available = cp_cruise.vl["CruiseControl"]["Cruise_On"] != 0
+    if self.car_fingerprint == CAR.FORESTER_2022:
+      ret.cruiseState.enabled = cp_cam.vl["ES_DashStatus"]['Cruise_Activated'] != 0
+      ret.cruiseState.available = cp_cam.vl["ES_DashStatus"]['Cruise_On'] != 0
```
Comment (sshane) [inline selfdrive/car/subaru/carstate.py:58]: did we verify that these signals are good representations of the cruise state? See my concerns in this PR: https://github.com/commaai/openpilot/pull/25378

Comment (martinl) [reply selfdrive/car/subaru/carstate.py:58]: We are looking into disable disengage on gas press not working issue that is probably related to second case in https://github.com/commaai/openpilot/pull/25378

**Comment by martinl** [comment]
Forester 2022 route: `7fd1e4f3a33c1673|2022-12-07--20-16-23`

Screenshot from 2022-12-10 17-53-21

ES_Status Cruise_Activated signal does not have any issues with standstill, gas and brake press, I will update panda safety and openpilot PR-s

File: selfdrive/car/subaru/carstate.py (hunk 1 above)
Comment (martinl) [reply selfdrive/car/subaru/carstate.py]: We switched to using ES_Status Cruise_Activated which behaves identically to CruiseControl Cruise_Activated signal on older models. That also fixed no disengage on gas press

File: selfdrive/car/subaru/carstate.py [hunk 2]
```diff
@@ -53,9 +53,13 @@ def update(self, cp, cp_cam, cp_body):
...
-    cp_cruise = cp_body if self.car_fingerprint in GLOBAL_GEN2 else cp
-    ret.cruiseState.enabled = cp_cruise.vl["CruiseControl"]["Cruise_Activated"] != 0
-    ret.cruiseState.available = cp_cruise.vl["CruiseControl"]["Cruise_On"] != 0
+    if self.car_fingerprint == CAR.FORESTER_2022:
+      ret.cruiseState.enabled = cp_cam.vl["ES_Status"]['Cruise_Activated'] != 0
+      ret.cruiseState.available = cp_cam.vl["ES_DashStatus"]['Cruise_On'] != 0
+    else:
+      cp_cruise = cp_body if self.car_fingerprint in GLOBAL_GEN2 else cp
+      ret.cruiseState.enabled = cp_cruise.vl["CruiseControl"]["Cruise_Activated"] != 0
+      ret.cruiseState.available = cp_cruise.vl["CruiseControl"]["Cruise_On"] != 0
```
Comment (sshane) [inline selfdrive/car/subaru/carstate.py:62]: Is there no signal that we can use on all cars? If not, what other cars are going to be like this?

Comment (martinl) [reply selfdrive/car/subaru/carstate.py:62]: Currently only Forester 2022 is confirmed, Crosstrek Hybrid is possible candidate but I have not been able to get a route yet

**Comment by jnewb1** [comment]
b0cea2db2dd5474e|2023-07-21--12-18-50

successful angle controller based route!

**Comment by jnewb1** [comment]
I will open a new PR for the angle based cars (outback23, forester22, and probably many more) when we get the safety merged:
https://github.com/commaai/panda/pull/1533

**Pull request closed without merging**